Flaky BackupManagerImplTest test

### Description

There is a test in `BackupManagerImplTest` that causes some failures in the CI:

```
2025-12-09T17:08:38.5114316Z [OK:    57, KO:     1, SKIP:     0] Test failed: org.infinispan.server.core.backup.BackupManagerImplTest.testBackupAndRestoreMultipleContainers
2025-12-09T17:08:38.5117203Z java.lang.Error: java.util.concurrent.ExecutionException: org.infinispan.commons.CacheConfigurationException: ISPN000502: Error while persisting global configuration state
2025-12-09T17:08:38.5120206Z    at org.infinispan.functional.FunctionalTestUtils.await(FunctionalTestUtils.java:72)
2025-12-09T17:08:38.5122365Z    at org.infinispan.server.core.backup.BackupManagerImplTest.testBackupAndRestoreMultipleContainers(BackupManagerImplTest.java:396)
2025-12-09T17:08:38.5124096Z    at java.base/java.lang.reflect.Method.invoke(Method.java:565)
2025-12-09T17:08:38.5125502Z    at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
2025-12-09T17:08:38.5127481Z    at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:230)
2025-12-09T17:08:38.5128933Z    at org.infinispan.commons.test.TestNGLongTestsHook.run(TestNGLongTestsHook.java:17)
2025-12-09T17:08:38.5130432Z    at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:242)
2025-12-09T17:08:38.5132152Z    at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:56)
2025-12-09T17:08:38.5133562Z    at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
2025-12-09T17:08:38.5134916Z    at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
2025-12-09T17:08:38.5137213Z    at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
2025-12-09T17:08:38.5138426Z    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
2025-12-09T17:08:38.5139797Z    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
2025-12-09T17:08:38.5140948Z    at java.base/java.lang.Thread.run(Thread.java:1474)
2025-12-09T17:08:38.5142860Z Caused by: java.util.concurrent.ExecutionException: org.infinispan.commons.CacheConfigurationException: ISPN000502: Error while persisting global configuration state
2025-12-09T17:08:38.5144855Z    at java.base/java.util.concurrent.CompletableFuture.wrapInExecutionException(CompletableFuture.java:345)
2025-12-09T17:08:38.5146322Z    at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:440)
2025-12-09T17:08:38.5147619Z    at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2117)
2025-12-09T17:08:38.5148887Z    at org.infinispan.functional.FunctionalTestUtils.await(FunctionalTestUtils.java:70)
2025-12-09T17:08:38.5149808Z    ... 13 more
2025-12-09T17:08:38.5150886Z Caused by: org.infinispan.commons.CacheConfigurationException: ISPN000502: Error while persisting global configuration state
2025-12-09T17:08:38.5153027Z    at org.infinispan.globalstate.impl.OverlayLocalConfigurationStorage.persistConfigurations(OverlayLocalConfigurationStorage.java:199)
2025-12-09T17:08:38.5155075Z    at org.infinispan.globalstate.impl.OverlayLocalConfigurationStorage.storeCaches(OverlayLocalConfigurationStorage.java:164)
2025-12-09T17:08:38.5157031Z    at org.infinispan.globalstate.impl.OverlayLocalConfigurationStorage.lambda$createCache$0(OverlayLocalConfigurationStorage.java:95)
2025-12-09T17:08:38.5158899Z    at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:667)
2025-12-09T17:08:38.5160305Z    at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:503)
2025-12-09T17:08:38.5161915Z    at java.base/java.util.concurrent.ThreadPerTaskExecutor$TaskRunner.run(ThreadPerTaskExecutor.java:291)
2025-12-09T17:08:38.5163154Z    at java.base/java.lang.VirtualThread.run(VirtualThread.java:456)
2025-12-09T17:08:38.5165295Z Caused by: org.infinispan.commons.CacheConfigurationException: ISPN000508: Cannot rename file /tmp/infinispanTempFiles/BackupManagerImplTest/caches6331666435609748270.tmp to /tmp/infinispanTempFiles/BackupManagerImplTest/caches.xml
2025-12-09T17:08:38.5168199Z    at org.infinispan.globalstate.impl.OverlayLocalConfigurationStorage.persistConfigurations(OverlayLocalConfigurationStorage.java:196)
2025-12-09T17:08:38.5169561Z    ... 6 more
2025-12-09T17:08:38.5170186Z Caused by: java.nio.channels.OverlappingFileLockException
2025-12-09T17:08:38.5171175Z    at java.base/sun.nio.ch.FileLockTable.checkList(FileLockTable.java:229)
2025-12-09T17:08:38.5172339Z    at java.base/sun.nio.ch.FileLockTable.add(FileLockTable.java:123)
2025-12-09T17:08:38.5173316Z    at java.base/sun.nio.ch.FileChannelImpl.lock(FileChannelImpl.java:1719)
2025-12-09T17:08:38.5174341Z    at java.base/java.nio.channels.FileChannel.lock(FileChannel.java:1223)
2025-12-09T17:08:38.5175364Z    at org.infinispan.commons.util.Util.renameTempFile(Util.java:956)
2025-12-09T17:08:38.5177088Z    at org.infinispan.globalstate.impl.OverlayLocalConfigurationStorage.persistConfigurations(OverlayLocalConfigurationStorage.java:194)
2025-12-09T17:08:38.5178458Z    ... 6 more
```

The test creates multiple cache containers simultaneously. Some possibilities are:

* The different containers map to the same caches.xml file and cause lock failure.
* More concerning: creating multiple caches within the same container concurrently tries to write to the file concurrently and fails.

### Implementation ideas

_No response_

--- a/server/core/src/test/java/org/infinispan/server/core/backup/BackupManagerImplTest.java
+++ b/server/core/src/test/java/org/infinispan/server/core/backup/BackupManagerImplTest.java
@@ -426,15 +426,22 @@ public class BackupManagerImplTest extends AbstractInfinispanTest {
 
    private Map<String, DefaultCacheManager> createManagerMap(String... containers) {
       return Arrays.stream(containers)
-            .collect(Collectors.toMap(Function.identity(), name -> createManager()));
+            .collect(Collectors.toMap(Function.identity(), this::createManager));
    }
 
    private DefaultCacheManager createManager() {
+      return createManager(UUID.randomUUID().toString());
+   }
+
+   private DefaultCacheManager createManager(String name) {
+      Path path = workingDir.toPath().resolve(name);
+      Util.recursiveFileRemove(path);
+      path.toFile().mkdir();
       GlobalConfigurationBuilder gcb = new GlobalConfigurationBuilder();
       gcb.cacheManagerName("default")
             .globalState().enable()
             .configurationStorage(ConfigurationStorage.OVERLAY)
-            .persistentLocation(workingDir.getAbsolutePath());
+            .persistentLocation(path.toFile().getAbsolutePath());
       return (DefaultCacheManager) TestCacheManagerFactory.newDefaultCacheManager(true, gcb, null);
    }
 

